Makefile

```diff
@@ -37,37 +34,81 @@ clean:: Makefile.coq.lib Makefile.coq
 	find . -type f \( -name '*~' -o -name '*.aux' -o -name '.lia.cache' -o -name '.nia.cache' \) -delete
 	rm -f Makefile.coq.lib Makefile.coq.lib.conf Makefile.coq Makefile.coq.conf _CoqProject
 
+EXTERNAL_DEPENDENCIES?=
+EXTERNAL_COQUTIL?=
+EXTERNAL_BEDROCK2?=
+
 COQUTIL_FOLDER := bedrock2/deps/coqutil
 BEDROCK2_FOLDER := bedrock2/bedrock2
 
+# Note: make does not interpret "\n", and this is intended
+DEPFLAGS_COQUTIL_NL=-Q $(COQUTIL_FOLDER)/src/coqutil coqutil\n
+DEPFLAGS_BEDROCK2_NL=-Q $(BEDROCK2_FOLDER)/src/bedrock2 bedrock2
+CURFLAGS_NL=-R src/Rupicola Rupicola\n-arg -w\n-arg -unexpected-implicit-declaration,-deprecated-ident-entry\n
+DEPFLAGS_NL=
+
+ifneq ($(EXTERNAL_COQUTIL),1)
+DEPFLAGS_NL+=$(DEPFLAGS_COQUTIL_NL)
+endif
+
+ifneq ($(EXTERNAL_BEDROCK2),1)
+DEPFLAGS_NL+=$(DEPFLAGS_BEDROCK2_NL)
+endif
+
+# If we get our dependencies externally, then we should not bind the local versions of things
+ifneq ($(EXTERNAL_DEPENDENCIES),1)
+ALLDEPFLAGS_NL=$(CURFLAGS_NL)$(DEPFLAGS_NL)
+else
+ALLDEPFLAGS_NL=$(CURFLAGS_NL)
+endif
+
+ifneq ($(EXTERNAL_DEPENDENCIES),1)
+
+ifneq ($(EXTERNAL_COQUTIL),1)
+bedrock2: coqutil
+install: install_coqutil
+endif
+
+ifneq ($(EXTERNAL_BEDROCK2),1)
+install: install_bedrock2
+all: bedrock2
+deps: bedrock2
+%.vo: bedrock2
+endif
+
+endif
+
 coqutil:
-	$(MAKE) --no-print-directory -C $(COQUTIL_FOLDER) 
+	$(MAKE) --no-print-directory -C $(COQUTIL_FOLDER)
 
 clean-coqutil:
 	$(MAKE) --no-print-directory -C $(COQUTIL_FOLDER) clean
 
-bedrock2: coqutil
+install-coqutil:
+	$(MAKE) --no-print-directory -C $(COQUTIL_FOLDER) install
+
+bedrock2:
 	$(MAKE) --no-print-directory -C $(BEDROCK2_FOLDER) noex
 
 clean-bedrock2:
 	$(MAKE) --no-print-directory -C $(BEDROCK2_FOLDER) clean
 
-deps: bedrock2
+install-bedrock2:
+	$(MAKE) --no-print-directory -C $(BEDROCK2_FOLDER) install_bedrock2
 
 cleanall: clean clean-coqutil clean-bedrock2
 
 %.vo: deps Makefile.coq
-	+make -f Makefile.coq $@
+	+$(MAKE) -f Makefile.coq $@
+
+install: Makefile.coq
+	+$(MAKE) -f Makefile.coq $@
 
-_CoqProject: Makefile
-	@echo "-R src/Rupicola Rupicola" > $@
-	@echo "-Q bedrock2/bedrock2/src/bedrock2 bedrock2" >> $@
-	@echo "-Q bedrock2/deps/coqutil/src/coqutil coqutil" >> $@
-	@echo "-arg -w" >> $@
-	@echo "-arg -unexpected-implicit-declaration,-deprecated-ident-entry" >> $@
+_CoqProject:
+	@printf -- '$(ALLDEPFLAGS_NL)' > _CoqProject
```

Comment:
This repo has a checked-in _CoqProject (whereas bedrock2 doesn't check it in), did you test if you still get the same when you regenerate it?

Reply:
Maybe we should just remove the checked-in _CoqProject?

Reply:
The diff is
```diff
diff --git a/_CoqProject b/_CoqProject
index a939e70..67def96 100644
--- a/_CoqProject
+++ b/_CoqProject
@@ -1,5 +1,5 @@
 -R src/Rupicola Rupicola
--Q bedrock2/bedrock2/src/bedrock2 bedrock2
--Q bedrock2/deps/coqutil/src/coqutil coqutil
 -arg -w
 -arg -unexpected-implicit-declaration,-deprecated-ident-entry
+ -Q bedrock2/deps/coqutil/src/coqutil coqutil
+ -Q bedrock2/bedrock2/src/bedrock2 bedrock2
```
btw

Reply:
_CoqProject removed